packages/core/src/modules/credentials/formats/indy/IndyCredentialFormatService.ts

```diff
@@ -89,7 +90,12 @@ export class IndyCredentialFormatService extends CredentialFormatService {
       throw new AriesFrameworkError('Missing payload in createProposal')
     }
 
-    const attachment: Attachment = this.getFormatData(options.credentialFormats.indy?.payload, formats.attachId)
+    // Use class instance instead of interface, otherwise this causes interoperability problems
+    let proposal = new CredPropose(options.credentialFormats.indy?.payload)
```

Comment:
Would be good to also validate the proposal class instance using `MessageValidator.validate` (and add an exception test)